### DISCUSSION TITLE
New Image Future is a downgrade?

### DISCUSSION THREAD

**Discussion by DanielOberlechner** [vercel/next.js#38126] [Help]
Hi guys,
I became today the email that a new nextjs version was released and there is a new Image Component in the Future branch.
I mean this one:
[NextJS Site](https://nextjs.org/docs/api-reference/next/future/image)
Changes are listed as the following:
Renders a single <img> without <div> or <span> wrappers
Adds support for canonical style prop
Removes layout, objectFit, and objectPosition props in favor of style or className
Removes IntersectionObserver implementation in favor of [native lazy loading](https://caniuse.com/loading-lazy-attr)
Removes loader config in favor of [loader](https://nextjs.org/docs/api-reference/next/future/image#loader) prop
Note: there is no fill mode so width & height props are required
Note: the [onError](https://nextjs.org/docs/api-reference/next/future/image#onerror) prop might behave differently

That now it is only compiled to one img tag is in my view a profit.
But that they did remove layout, objectfit and objectposition is kinda a disadvantage in my opinion.

For example I used to render a picture of myself as a quadrate the following code:
```javascript
                <Image src={image} classes={"w-full xl:h-144 relative image-zoom img-border"} layout={"responsive"} width={16} height={9} objectFit={"contain"} objectPosition={"center"} alt={""}/>
```

But with the new tag everything changes and in my opinion not into the positive.
Why should I use the style tags to define the object properties?
I think the old way of styling it was more convenient.

I also played a lot with the layout properties around when the Image did not look like I expected to do it.
How would I even do this with css?
What would be the same CSS properties for the layout ones?

Am I the only one who thinks so and cannot see the advantages or are there other people who see it like me?

Kind regards,
Daniel Oberlechner

**Comment by sandren** [comment]
The new `<Image />` components plays really nicely with Tailwind utility classes and any other method of styling.

So you can add things like `className="object-cover w-full h-full"` and `className="object-contain w-full h-full"` to resize the image according to the container or set the aspect ratio directly on the image itself like `aspect-video` for a 16/9 wide aspect ratio.

Alternatively you could do the same thing with the `style` prop and inline styles although it's a bit more verbose. 🙂

**Comment by DanielOberlechner** [comment]
Yeah I understand. I used tailwind previously too and it worked then as well.
There was no need for tailwindcss aspect ratio, because with the old Image tag and a few settings I could get the aspect ratio from 16:9 as well without the special tailwindcss plugin ...

For me it was easier to experiment with the Image Layouts with the old Image tag. There I could rush a few settings threw and I saw immediately what changed ...

**Reply by sandren** [reply]
> Yeah I understand. I used tailwind previously too and it worked then as well. There was no need for tailwindcss aspect ratio, because with the old Image tag and a few settings I could get the aspect ratio from 16:9 as well without the special tailwindcss plugin ...
> 
> For me it was easier to experiment with the Image Layouts with the old Image tag. There I could rush a few settings threw and I saw immediately what changed ...

Things will immediately change based on `className` and `style` props as well just like the old `objectFit` and `objectPosition`.

Perhaps it might be helpful to think of styling the new `<Image />` component as if it were a regular `<img />` element since there's no more wrapper elements. 🙂

Also the Tailwind aspect ratio plugin is no longer required as it is a native part of Tailwind v3: https://tailwindcss.com/docs/aspect-ratio

**Comment by GabenGar** [comment]
The problem with the old `Image` is it relies on hardcoded markup with inline styles, which made it, at the very best, very difficult to implement in an existing codebase.

**Comment by hjaber** [comment]
The new image component is a big improvement. Rendering a single `<img>` it is much more manageable to style and you don't have to do strange things to remove padding such as `font-size: 0px`. It took some debugging to solve a layout shift but it was an easy transition.
